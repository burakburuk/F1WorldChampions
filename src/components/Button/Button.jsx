
import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledButton = styled.button`
    border: none;
    padding: 10px 20px;
    text-align: center;
    font-size: 14px;
    cursor: pointer;
    color: ${props => (props.primary ? '#51a4de' : '#ffffff')};
    border-width: 1px;
    border-style: solid;
    border-color: ${props => (props.primary ? '#51a4de' : '#ece9e9')};
    background-color: ${props => (props.primary ? '#ffffff' : '#ece9e9')};

    &:hover {
        background-color: ${props => (props.primary ? '#3e82b1' : '')};
        color: #ffffff;
    }
`;

const Button = props => (
    <StyledButton
        primary={props.primary}
        onClick={props.onClick} >
        {props.children}
    </StyledButton>
);

Button.propTypes = {
    primary: PropTypes.bool,
    onClick: PropTypes.func,
    children: PropTypes.string.isRequired,
};

export default Button;
