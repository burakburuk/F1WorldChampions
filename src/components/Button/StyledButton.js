
import styled from 'styled-components';

const StyledButton = styled.button`
    border: none;
    padding: 10px 20px;
    text-align: center;
    cursor: pointer;
    color: ${props => (props.primary ? '#51a4de' : '#ffffff')};
    border-width: 1px;
    border-style: solid;
    border-color: ${props => (props.primary ? '#51a4de' : '#ece9e9')};
    background-color: ${props => (props.primary ? '#ffffff' : '#ece9e9')};

    &:hover {
        background-color: ${props => (props.primary ? '#3e82b1' : '')};
        color: #ffffff;
    }
`;

export default StyledButton;
