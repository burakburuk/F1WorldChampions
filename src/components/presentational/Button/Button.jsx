
import React from 'react';
import PropTypes from 'prop-types';

const Button = (props) => {
    const classNames = `button ${props.className} ${getColor(props.color)}`;
    return (
        <button
            className={classNames}
            onClick={props.onClick} >
            {props.children}
        </button>
    );
};

function getColor(color) {
    switch (color) {
    case 'primary':
        return 'button-primary';
    case 'disabled':
        return 'button-gray';
    default:
        return 'button-primary';
    }
}

Button.propTypes = {
    className: PropTypes.string,
    color: PropTypes.string,
    onClick: PropTypes.func,
    children: PropTypes.string.isRequired,
};

export default Button;
