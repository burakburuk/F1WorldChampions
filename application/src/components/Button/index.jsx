
import React from 'react';
import PropTypes from 'prop-types';
import StyledButton from './StyledButton';

const Button = props => (
    <StyledButton
        primary={props.primary}
        onClick={props.onClick} >
        {props.children}
    </StyledButton>
);

Button.propTypes = {
    primary: PropTypes.bool,
    onClick: PropTypes.func,
    children: PropTypes.string.isRequired,
};

export default Button;
