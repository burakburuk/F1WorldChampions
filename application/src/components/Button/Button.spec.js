import React from 'react';
import { shallow } from 'enzyme';
import Button from '.';

it('should match with snapshot', () => {
    const wrapper = shallow(<Button>Test button</Button>);
    expect(wrapper).toMatchSnapshot();
});
