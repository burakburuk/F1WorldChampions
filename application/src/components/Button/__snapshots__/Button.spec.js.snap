// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<styled.button>
  Test button
</styled.button>
`;
